bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DraftsAndSubmissionsImpl.java

```diff
@@ -0,0 +1,348 @@
+    private static final String TOOLTIP = "Open";
+    private static final String OPERATION = "operation";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DraftsAndSubmissionsImpl.class);
+
+    private SimpleDateFormat dateFormatter = new SimpleDateFormat("hh:mm:ss dd-M-yy");
+
+    @Self
+    @Required
+    private SlingHttpServletRequest request;
+
+    @SlingObject
+    private ResourceResolver resolver;
+
+    @OSGiService
+    private DraftService draftService;
+
+    @OSGiService
+    private SubmitService submitService;
+
+    @OSGiService
+    private PendingSignService pendingSignService;
+
+    @OSGiService
+    private OperationManager operationManager;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    @Inject
+    @Default(values = "DRAFT")
+    private String type;
+
+    @Override
+    protected String defaultTitle() {
+        return "Drafts";
+    }
+
+    @Override
+    public String getType() {
+        return type;
+    }
+
+    private PortalLister.Item getItem(final String formPath, final TypeEnum typeEnum, final String id, final String timeInfo) {
+        String title = "<Non Existent Form>";
+        String description = null;
+        String thubmnail = null;
+        String formLink = null;
+
+        ResourceResolver resourceResolver = request.getResourceResolver();
+        String formAssetPath = GuideUtils.convertGuideContainerPathToFMAssetPath(formPath);
+        Resource formAssetResource = resourceResolver.getResource(formAssetPath);
+        if (formAssetResource != null) {
+            AdaptiveFormAsset asset = formAssetResource.adaptTo(AdaptiveFormAsset.class);
+            title = asset.getTitle();
+            description = asset.getDescription();
+            thubmnail = asset.getThumbnailPath();
+            formLink = asset.getRenderLink();
+
+            if (TypeEnum.DRAFT == typeEnum) {
+                try {
+                    URI newLink = new URIBuilder(formLink).addParameter("dataRef", String.format(DRAFT_LINK, id)).build();
+                    formLink = newLink.toString();
+                } catch (URISyntaxException e) {
+                    LOGGER.error("[FP] Could not parse render link", e);
+                }
+            }
+        }
+
+        PortalListerImpl.Item item = new PortalListerImpl.Item();
+        item.setTitle(title);
+        item.setDescription(description);
+        item.setFormLink(formLink);
+        item.setTooltip(TOOLTIP);
+        item.setFormThumbnail(thubmnail);
+        item.setId(id);
+        item.setOperations(operationManager.getOperationList(typeEnum));
+        item.setTimeInfo(timeInfo);
+        return item;
+    }
+
+    private Boolean isOperationCall() {
+        String operationName = getOperationName();
+        return StringUtils.isNotEmpty(operationName);
+    }
+
+    @Override
+    public Operation.OperationResult getOperationResult() {
```

Comment:
This is because sling model is re-used for executing operations.
Other way discussed would be to use a different servlet. Currently we're going forward with re-using model as discussed.